custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -1164,3 +1172,71 @@ def create_custom_inference_model_folder(code_dir, output_dir):
         fp.write(readme)
     if files_in_output & copied_files:
         print("Files were overwritten: {}".format(files_in_output & copied_files))
+
+
+def get_default_parameter_values(model_metadata: Dict) -> Dict[str, Union[int, float, str]]:
+    """Retrieve default parameter values from the hyperparameter section of model-metadata.yaml.
+    When `default` is provided, return the default value.
+    When `default` is not provided:
+        - Return `min` value if it is a numeric parameter.
+        - Return an emtpy string if it is a string parameter.
+        - Return the first allowed value if it is the select parameter.
+        - Return the default value of the first component parameter (sorted by parameter type) if it is a multi
+          parameter.
+    """
+
+    def _get_default_numeric_param_value(param_config: Dict) -> Union[int, float]:
+        """Get default value of numeric parameter."""
+        param_default_value = param_config.get("default")
+        if param_default_value is not None:
+            return param_default_value
+        else:
+            return param_config["min"]
+
+    def _get_default_string_param_value(param_config: Dict) -> str:
+        """Get default value of string parameter."""
+        param_default_value = param_config.get("default")
+        if param_default_value is not None:
+            return param_default_value
+        else:
+            return ""
+
+    def _get_default_select_param_value(param_config: Dict) -> str:
+        """Get default value of select parameter."""
+        param_default_value = param_config.get("default")
+        if param_default_value is not None:
+            return param_default_value
+        else:
+            return param_config["values"][0]
+
+    def _get_default_multi_param_value(param_config: Dict) -> str:
+        """Get default value of multi parameter."""
+        param_default_value = param_config.get("default")
+        if param_default_value is not None:
+            return param_default_value
+        else:
+            param_values = param_config["values"]
+            if param_values:
```

Comment:
With the existing schema design, it is possible that `values` is an empty dictionary. The highlight logic is adapted to the following schema design.
```
PARAM_MULTI_T = t.Dict(
    {
        t.Key('name'): t.String(max_length=PARAM_NAME_MAX_LENGTH),
        t.Key('type'): t.Enum('multi'),
        t.Key('values'): t.Dict(
            {
                t.Key('int', optional=True): t.Dict({t.Key('min'): t.Int, t.Key('max'): t.Int,}),
                t.Key('float', optional=True): t.Dict(
                    {t.Key('min'): t.Float, t.Key('max'): t.Float,}
                ),
                t.Key('select', optional=True): t.Dict(
                    {
                        t.Key('values'): t.List(
                            t.String(max_length=PARAM_SELECT_VALUE_MAX_LENGTH, allow_blank=False),
                            min_length=1,
                            max_length=PARAM_SELECT_NUM_VALUES_MAX_LENGTH,
                        ),
                    }
                ),
            }
        ),
        t.Key('default', optional=True): t.Or(
            t.Int, t.Float, t.String(max_length=PARAM_SELECT_VALUE_MAX_LENGTH)
        ),
    }
)
```

Reply:
we should just insist that values is not an empty dictionary.